docker/docker_client.go

```diff
@@ -996,7 +997,18 @@ func (c *dockerClient) fetchManifest(ctx context.Context, ref dockerReference, t
 	if err != nil {
 		return nil, "", err
 	}
-	return manblob, simplifyContentType(res.Header.Get("Content-Type")), nil
+	// Extra check for the content type in the manifest
+	if contentType == "text/plain" {
```

Comment:
- It is fundamentally unsound to override an externally-supplied MIME type of some data by parsing the data in another way and trusting its contents. The whole point of having the MIME type separate is that it directs how the data is to be interpreted. I’m unhappy about such special cases, and I’m especially unhappy about adding them in entirely new use cases where we could just choose not to do that. Why is this necessary here?
- The `docker` transport has no business interpreting image contents this way; that’s the responsibility of the generic code.
- … and the generic code _already has_ a heuristic fallback for `text/plain`; this would override/break that.